### PR TITLE
Update hashes now that GSI-B is working for EnVar

### PR DESCRIPTION
**Pull request by CoryMartin-NOAA**
SABER develop will now only build with GSIbec 1.2.1. This PR brings all the repos to `develop` and gsibec to `1.2.1`.

### CODE REVIEW

**Review by guillaumevernieres — APPROVED** [review]

**Comment by CoryMartin-NOAA** [comment]
Will merge once CI runs and hopefully passes

**Review by danholdaway — APPROVED** [review]

**Review by DavidNew-NOAA — APPROVED** [review]

**Comment by CoryMartin-NOAA** [comment]
The two tests will fail until @DavidNew-NOAA 's workflow PR gets merged in. I will merge this in now.

**Pull request merged**
